tools/compat/getopt.rs

```diff
@@ -0,0 +1,28 @@
+cfg_if::cfg_if! {
+    if #[cfg(target_os = "windows")] {
```

Comment:
Why is this `cfg_if`? It could just be a `mod` with a `cfg` on it and then re-exports.

Reply:
Hm, I see the other compat modules do this. It's fine, just seemed weird.